packages/web/src/components/notification/Notifications/NotificationTile.tsx

```diff
@@ -12,11 +19,24 @@ type NotificationTileProps = {
   onClick?: ReactEventHandler
   // When `true` disable :active transforms
   disabled?: boolean
+  // When `true` do not close notification panel onClick
+  disableClosePanel?: boolean
 }
 
 export const NotificationTile = (props: NotificationTileProps) => {
-  const { notification, onClick, children, disabled } = props
+  const { notification, onClick, children, disabled, disableClosePanel } = props
   const { isRead } = notification
+  const dispatch = useDispatch()
+
+  const handleClick: MouseEventHandler = useCallback(
+    event => {
+      onClick?.(event)
+      if (!disableClosePanel) {
```

Comment:
This works, but after thinking about it, what if we don't close the `NotificationsPanel` on clicks, but rather on navigations? The goal is to close it whenever the user navigates away, right? I guess we'd need both as we probably want to close it on Twitter modal popups? I'm a bit uninformed here for sure.

Reply:
yeah this is good thinking for sure! yeah do we want to close in on twitter? we don't currently which i think makes sense. so def agree with the idea that it should be on navigates, how would we do that for the complex interaction of `click follow notif with multiple follows -> opens user list modal -> click on a user goes to profile and closes notification?` just have an event listener for url changes in panel?

Reply:
Ah if we don't currently then no - I'm a little hazy on the ins/outs of when we close it currently sorry! All the route changes should be going through the store, right? Maybe a saga listens for route pushes and put an action to hide notif panel? Seems like a misuse of sagas, but maybe it's fine 😁  Could also be a reducer but i think we slice those up so not sure a notif panel reducer would see the route actions

Reply:
no worries youre thinking is spot on. ill think about things more and see if i come to a sensible way of handling this!